### PR TITLE
Feature/upgrade leaflet

### PR DESCRIPTION
**Pull request by Daniel-KM**
This is an upgrade to the last version of leaflet and the associate plugins.

### CODE REVIEW

File: asset/js/mapping-block.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ mappingMaps.each(function() {
         'Satellite': L.tileLayer.provider('Esri.WorldImagery'),
         'Terrain': L.tileLayer.provider('Esri.WorldShadedRelief')
     };
-    var noOverlayLayer = L.tileLayer.canvas();
+    var noOverlayLayer = new L.GridLayer();
```
Comment (jimsafley) [inline asset/js/mapping-block.js:46]: What's the rationale for switching from `L.tileLayer.canvas()` to `new L.GridLayer()`?

File: asset/js/mapping-collecting-form.js [hunk 2]
```diff
@@ -13,8 +13,8 @@ $(document).ready(function() {
         }).addTo(map);
         map.setView([20, 0], 2);
 
-        map.addControl(new L.Control.GeoSearch({
-            provider: new L.GeoSearch.Provider.OpenStreetMap(),
+        map.addControl(new window.GeoSearch.GeoSearchControl({
+            provider: new window.GeoSearch.OpenStreetMapProvider,
```
Comment (jimsafley) [inline asset/js/mapping-collecting-form.js:17]: What's the rationale for switching from `L.Control.GeoSearch()` to `window.GeoSearch`?

**Review by jimsafley — COMMENTED** [review]
Thank you for the vendor updates and block layout fix. We're not going to merge the CSV commits into master with this PR.

File: asset/js/mapping-block.js (hunk 1 above)
Comment (Daniel-KM) [reply asset/js/mapping-block.js:46]: It was remove some years ago: https://github.com/Leaflet/Leaflet/commit/64ca1b33b823409901a7b7a55bc6b4bf226f56b4

File: asset/js/mapping-collecting-form.js (hunk 2 above)
Comment (Daniel-KM) [reply asset/js/mapping-collecting-form.js:17]: Geosearch is now an independant plugin, that can work without Leaflet, so it is attached by default to the window. Some more works in the javascript is needed to integrate it in the Leaflet, but it is useless.

**Comment by Daniel-KM** [comment]
I'm going to rebase the pr on upstream.

**Comment by Daniel-KM** [comment]
It was rebased.

**Comment by jimsafley** [comment]
Again, thank you.

**Pull request merged**